### PR TITLE
Add missing db providers to db.system

### LINKED ISSUE #882: [Instrumentation.EntityFrameworkCore] Extend `db.system` tags.
# Issue with OpenTelemetry.Instrumentation.EntityFrameworkCore

List of [all OpenTelemetry NuGet
packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are
using (e.g. `OpenTelemetry 1.0.2`):

* Current main - after OpenTelemetry.Instrumentation.EntityFrameworkCore 1.0.0-beta.3

Runtime version (e.g. `net462`, `net48`, `netcoreapp3.1`, `net6.0` etc. You can
find this information from the `*.csproj` file):

All of them.

**Is this a feature request or a bug?**

* [x] Feature Request
* [ ] Bug

**What is the expected behavior?**

All packages from https://learn.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli
are mapped to the appropriate `db.system` tags based on https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md?plain=1#L67-L120

**What is the actual behavior?**

Only part of the providers are fully handled. Other are mapped to other balue:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs#L99-L123

### PR DESCRIPTION
**Pull request by mrblonde91**
Fixes #882, outstanding question is the ibm provider as it can represent two different dbs. I'm inclined to favour putting it under other rather than a generic banner.
## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [x] Design discussion issue #882

### CODE REVIEW

**Comment by Kielek** [comment]
> Labeled spanner as `gcp_spanner` as it's not currently got a naming scheme.

Please. consider to create PR similar to https://github.com/open-telemetry/opentelemetry-specification/pull/2718. Bassed on other values, I suppose just `spanner` will be better value.

**Comment by mrblonde91** [comment]
> > Labeled spanner as `gcp_spanner` as it's not currently got a naming scheme.
> 
> Please. consider to create PR similar to [open-telemetry/opentelemetry-specification#2718](https://github.com/open-telemetry/opentelemetry-specification/pull/2718). Bassed on other values, I suppose just `spanner` will be better value.

Opened, omitted efcore in memory and filecontext core cause they're more for prototyping purposes to some degree imho. https://github.com/open-telemetry/opentelemetry-specification/pull/3105

File: src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs [hunk 1]
```diff
@@ -105,18 +105,52 @@ public override void OnCustom(string name, Activity activity, object payload)
...
+                            case "Google.Cloud.EntityFrameworkCore.Spanner":
+                                activity.AddTag(AttributeDbSystem, "spanner");
+                                break;
+                            case "Teradata.EntityFrameworkCore":
+                                activity.AddTag(AttributeDbSystem, "teradata");
+                                break;
+                            case "IBM.EntityFrameworkCore":
+                            case "IBM.EntityFrameworkCore-lnx":
+                            case "IBM.EntityFrameworkCore-osx":
+                                activity.AddTag(AttributeDbSystem, "ibm");
```
Comment (mrblonde91) [inline src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs:152]: This one is an open question imho. Id'ing it as an ibm system seems better than defaulting outright to other_sql.

**Comment by mrblonde91** [comment]
@Kielek Pr is now merged to the specs so the naming is matching. I've decided to omit the ibm ones as it's not possible to differentiate between db2 or informix. If we want to include, it needs to be a more generic banner.

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.EntityFrameworkCore/CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,10 @@ Released 2023-Jan-25
 * Added support to `EnrichWithIDbCommand`
   ([#868](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/868))
 
+* Map missing dbs to system.db:
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.EntityFrameworkCore/CHANGELOG.md:23]:
```suggestion
* Map missing dbs to db.system:
```

**Review by utpilla — APPROVED** [review]

**Pull request merged**